apps/web/src/components/activity/ExecutionDetailsStepHeader.tsx

```diff
@@ -0,0 +1,219 @@
+import styled from 'styled-components';
+
+import { colors, Text } from '../../design-system';
+import {
+  Chat,
+  CheckCircle,
+  Clicked,
+  Digest,
+  ErrorIcon,
+  InApp,
+  Mail,
+  Read,
+  Received,
+  Seen,
+  Sent,
+  Sms,
+  Timer,
+} from '../../design-system/icons';
+
+const StepName = styled(Text)<{ theme: string }>`
+  color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B40)};
+  font-size: 16px;
+  font-weight: 700;
+  line-height: 17px;
+  padding-bottom: 5px;
+  padding-top: 0;
+  text-transform: capitalize;
+`;
+
+const StepDetails = styled(Text)<{ theme: string }>`
+  color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B60)};
+  font-size: 12px;
+  line-height: 16px;
+  padding: 0;
+`;
+
+const StepDate = styled(Text)<{ theme: string }>`
+  color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.B30 : colors.B60)};
+  font-size: 12px;
+  line-height: 16px;
+  padding: 3px 0 0;
+`;
+
+const FailedStepDetails = styled(StepDetails)`
+  color: ${colors.error};
+`;
+
+const SuccessStepDetails = styled(StepDetails)`
+  color: ${colors.success};
+`;
+
+const LogoWrapper = styled(Container)`
+  padding: 10px 10px 0 0;
+`;
+
+const StepActionOutcomeWrapper = styled(Container)`
+  align-items: center;
+  display: flex;
+  flex-flow: column;
+  justify-content: center;
+  margin: 0;
+  padding: 15px 0 0;
+`;
+
+const StepActionOutcomeName = styled(Text)`
+  color: ${colors.B60};
+  font-size: 12px;
+  line-height: 16px;
+  padding-top: 5px;
+`;
+
+const getLogoStyledComponentByStepStatus = (status, type) => {
+  if (type === StepTypeEnum.DIGEST) {
+    return Digest;
+  }
+
+  if (type === StepTypeEnum.DELAY) {
+    return Timer;
+  }
+
+  if (status === ExecutionDetailsStatusEnum.SUCCESS) {
+    return CheckCircle;
+  }
+
+  if (type === ChannelTypeEnum.EMAIL) {
+    return Mail;
+  }
+  if (type === ChannelTypeEnum.SMS) {
+    return Sms;
+  }
+  if (type === ChannelTypeEnum.IN_APP) {
+    return InApp;
+  }
+  if (type === ChannelTypeEnum.CHAT) {
+    return Chat;
+  }
+
+  // TODO: Catch all logo
+  return Mail;
+};
+
+const getColorByStatus = (status) => {
+  if (status === ExecutionDetailsStatusEnum.SUCCESS) {
+    return colors.success;
+  }
+
+  return undefined;
+};
+
+const StepLogo = ({ status, type }) => {
+  const Logo = getLogoStyledComponentByStepStatus(status, type);
+  const color = getColorByStatus(status);
+
+  return (
+    <LogoWrapper>
+      <Logo height="32px" width="32px" color={color} />
+      {status === ExecutionDetailsStatusEnum.FAILED && (
+        <ErrorIcon
+          height="15px"
+          width="15px"
+          color={colors.error}
+          style={{ position: 'relative', left: '20px', top: '-15px' }}
+        />
+      )}
+    </LogoWrapper>
+  );
+};
+
+const generateDetailByStepAndStatus = (status, step) => {
+  if (status === ExecutionDetailsStatusEnum.SUCCESS) {
+    return 'Success! The notification has been delivered.';
```

Comment:
I feel that we should do a different success message for delay and digest. Something like 'Finished digesting events!' or something else haha 
What do you think?

Reply:
That's a good point. Let me give it a try.